controllers/storagecluster/cephcluster.go

```diff
@@ -530,13 +528,15 @@ func getNetworkSpec(sc ocsv1.StorageCluster) rookCephv1.NetworkSpec {
 	// respect both the old way and the new way for enabling HostNetwork
 	networkSpec.HostNetwork = networkSpec.HostNetwork || sc.Spec.HostNetwork
 
-	// Enable the msgr2 port always
-	if networkSpec.Connections == nil {
-		networkSpec.Connections = &rookCephv1.ConnectionsSpec{
-			RequireMsgr2: true,
+	// Enable the msgr2 port always if it's not an external cluster
+	if !sc.Spec.ExternalStorage.Enable {
+		if networkSpec.Connections == nil {
```

Comment:
a small suggestion to simplify the flow:
```suggestion
		if networkSpec.Connections == nil {
			networkSpec.Connections = &rookCephv1.ConnectionsSpec{}
		}
		networkSpec.Connections.RequireMsgr2 = true
```

Reply:
done